### PR TITLE
Adds a small note that talks about master and node versions matching

### PR DESCRIPTION
**Pull request by stevsmit**
No BZ. Follow up to https://github.com/openshift/openshift-docs/pull/39184

Preview: https://deploy-preview-39647--osdocs.netlify.app/openshift-enterprise/latest/architecture/control-plane#control-plane-and-node-host-compatibility

SME ack'd in private slack message: 

W. Trevor King  4:03 PM
Both of those hit all the points I'm interested in 

Please check for formatting.

### CODE REVIEW

**Comment by jeana-redhat** [comment]
Just what I pinged you about in Slack - otherwise LGTM

File: modules/architecture-machine-roles.adoc [hunk 1]
```diff
@@ -8,7 +8,9 @@
 
 [NOTE]
 ====
-The cluster also contains the definition for the bootstrap role. Because the bootstrap machine is used only during cluster installation, its function is explained in the cluster installation documentation.
+* The cluster also contains the definition for the bootstrap role. Because the bootstrap machine is used only during cluster installation, its function is explained in the cluster installation documentation.
+
+* The {product-title} version must match between control plane host and node host. For example, in a 4.9 cluster, all control plane hosts must be 4.9 and all nodes must be 4.9. In some cases, temporary mismatches during cluster upgrades might arise. For odd {product-title} minor versions, kubelet versions 0 or 1 minor versions behind the API server version are supported. For even {product-title} minor versions, kubelet versions 0, 1, or 2 minor versions behind the API server version are supported.
```
Comment (sferich888) [inline modules/architecture-machine-roles.adoc:13]: I wish we could define 'some cases' better. Is is possible to enumerate them? Is this for upgrades? Disaster Recover? How long can my cluster be in this state? What are the boundaries of support? 

IMO this wording is to ambgous and needs to be refined better, using technical limitations and boundaries. 

If we don't do this; we might as well just say we support version drift (as you have stated it for odd and even minor versions - as confusing as that is).

File: modules/architecture-machine-roles.adoc [hunk 2]
```diff
@@ -11,6 +11,32 @@
 The cluster also contains the definition for the bootstrap role. Because the bootstrap machine is used only during cluster installation, its function is explained in the cluster installation documentation.
 ====
 
+== Control plane and node host compatibility
+
+The {product-title} version must match between control plane host and node host. For example, in a 4.9 cluster, all control plane hosts must be 4.9 and all nodes must be 4.9.
+
+Temporary mismatches during cluster upgrades are acceptable. For example, when upgrading from {product-title} 4.8 to 4.9, some nodes will upgrade to 4.9 before others. Prolonged skewing of control plane hosts and node hosts might expose older computes to bugs and missing features. Users should resolve skewed control plane hosts and node hosts as soon as possible.
```
Comment (kmccarron-rh) [inline modules/architecture-machine-roles.adoc:18]: I'm having to pause a bit here -Is there a better way to express this concept?

**Review by kmccarron-rh — COMMENTED** [review]
One small nit, otherwise lgtm!

File: modules/architecture-machine-roles.adoc [hunk 3]
```diff
@@ -11,6 +11,32 @@
...
+| Odd {product-title} minor versions ^[1]^
+| Up to one version older
+
+| Even {product-title} minor versions ^[2]^
+| Up to two versions older
+|===
+[.small]
+--
+1. For example, {product-title} 4.5, 4.7, 4.9.
+2. For example, {product-title} 4.6, 4.9, 4.10.
```
Comment (kmccarron-rh) [inline modules/architecture-machine-roles.adoc:37]: Should this be 4.6, 4.8, 4.10?

**Pull request merged**